kanin/src/extract/state.rs

```diff
@@ -1,15 +1,18 @@
 //! Allows extracting app state.
 
+use std::convert::Infallible;
+
 use async_trait::async_trait;
 use derive_more::{Deref, DerefMut};
-use tracing::error;
 
-use crate::{error::ServerError, Extract, HandlerError, Request};
+use crate::{Extract, Request};
 
-/// `State` is an extractor helper struct that allows you to extract app state
-/// that has previously been added to the app through a call to [`crate::App::state`]
+/// `State` is an extractor helper struct that allows you to extract app state from the state type added in `App::new`.
+///
+/// This implements `Deref` and `DerefMut` to the inner type, but the inner type is also public so you can also just take ownership if you prefer.
 ///
-/// This implements `Deref` and `DerefMut` to the inner type.
+/// Any type that implements `From<&S>` where `S` is the app state given in `App::new` can be extracted via this type.
+/// These `From` implementations can be derived on a struct via `kanin::AppState`.
```

Comment:
```suggestion
/// These `From` implementations can be derived on a struct via `kanin::AppState`.
///
/// # Example
/// ```
/// # struct AppState;
/// async fn my_handler(State(app_state): State<AppState>) {
///     // ...
/// }
/// ```
```

It took me longer than I care to admit to realize that this is just the extractor. I think documentation could be stronger with some examples.